src/nodeViews/createReactNodeViewConstructor.tsx

```diff
@@ -207,10 +216,29 @@ export function createReactNodeViewConstructor(
       />
     );
 
+    if (contentDOM) {
+      // Store this node view's registration key on the contentDOM
+      // so it can be retrieved by its children.
+      contentDOM.dataset["registrationkey"] = key;
```

Comment:
Maybe we can make it clearer what this is and say `react-pm-key` or something?

Reply:
It feels like this key in some sense identifies this node, which makes me tempted to place it on `dom` instead of `contentDOM` even if that means descendants will have to traverse a little bit farther to find it. Does that make any sense?

Reply:
Maybe we should also assign this constant value to a symbol.

Reply:
I guess I mean to say "assign a symbol to this constant value".

Reply:
And by that I don't mean `Symbol`, just a variable. 😆

Reply:
Yes, that's great

Reply:
(I'm gonna leave it on contentDOM, though; if that's what ProseMirror says "is the DOM at that position", who am I to argue?)

Reply:
ProseMirror tells us the starting point for our search, but that need not be the same place as where we stash the key. I'm not sure we're arguing with ProseMirror at all here.

Reply:
I might be overthinking this, though, and certainly this is an internal detail we can change later. I think I am reaching out with my imagination towards a need to identify the nodes in the DOM for other reasons and feeling like the identity of a node is more in line with its DOM than its content DOM.